pkg/stanza/flush/flush.go

```diff
@@ -11,15 +11,16 @@ import (
 )
 
 type State struct {
+	FlushPeriod    time.Duration
```

Comment:
Important detail to note is that this struct is part of what is written to storage in order to ensure consistent behavior in the event of a restart. However, the flush period is part of the configuration of the receiver, which may change during a restart. By initializing it and then persisting it, we are ignoring a potential config update and inheriting the previous config setting.

Reply:
Then... I should not do this. Thanks for explaining. Will come with a different approach.

Reply:
Fixed differently.